docs/extensions.md

```diff
@@ -0,0 +1,697 @@
+
+```bash
+php ../php-src/build/gen_stub.php extension.stub.php
+```
+
+This script will generate a file named `extension_arginfo.h` that contains the necessary information for PHP to know how to define and call our function.
+
+#### Write the Bridge Between Go and C
+
+Now, we need to write the bridge between Go and C. Create a file named `extension.h` in your module directory with the following content:
+
+```c
+#ifndef _EXTENSION_H
+#define _EXTENSION_H
+
+void register_extension();
+
+#endif
+```
+
+Next, create a file named `extension.c` that will perform the following steps:
+
+* Include PHP headers;
+* Declare our new native PHP function `go_print()`;
+* Declare the extension metadata;
+* Define the `register_extension()` function that will register our extension with PHP.
+
+Let's start by including the required headers:
+
+```c
+#include <php.h>
+#include "extension.h"
+#include "extension_arginfo.h"
+
+// Contains symbols exported by Go
+#include "_cgo_export.h"
+
+static int (*original_php_register_internal_extensions_func)(void) = NULL;
+```
+
+For PHP to recognize our extension, we need to use [one of the function pointers](https://github.com/php/php-src/blob/d585a5609d9d9ad9390eeb8c3109b64ff02bd632/main/main.c#L90) used internally by PHP. To avoid losing the original pointer, we store it in an `original_php_register_internal_extensions_func` variable to call it when registering our extension.
+
+We then define our PHP function as a native language function:
+
+```c
+PHP_FUNCTION(go_print)
+{
+    if (zend_parse_parameters_none() == FAILURE) {
+        RETURN_THROWS();
+    }
+
+    go_print_something();
+}
+
+zend_module_entry ext_module_entry = {
+    STANDARD_MODULE_HEADER,
+    "ext_go",
+    ext_functions, /* Functions */
+    NULL,          /* MINIT */
+    NULL,          /* MSHUTDOWN */
+    NULL,          /* RINIT */
+    NULL,          /* RSHUTDOWN */
+    NULL,          /* MINFO */
+    "0.1.1",
+    STANDARD_MODULE_PROPERTIES
+};
+```
+
+In this case, our function takes no parameters and returns nothing. It simply calls the Go function we defined earlier, exported using the `//export` directive.
+
+Secondly, we define the extension's metadata in a `zend_module_entry` structure, such as its name, version, and properties. This information is necessary for PHP to recognize and load our extension. Note that `ext_functions` is an array of pointers to the PHP functions we defined, and it was automatically generated by the `gen_stub.php` script in the `extension_arginfo.h` file.
+
+Finally, we need to register our extension with PHP using the `register_extension()` functions (which we call in our Go code), but also taking care to call the original function pointer so as not to overwrite other extensions:
+
+```c
+PHPAPI int register_internal_extensions(void)
+{
```

Comment:
I suggest using helper functions provided by FrankenPHP even for the "manual" code. WDYT?

Reply:
Suits me fine as it showcases FrankenPHP benefits even when creating manually the extension!